plugins/csp-wms-overlays/src/Plugin.hpp

```diff
@@ -0,0 +1,135 @@
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#ifndef CSP_WMS_OVERLAYS_PLUGIN_HPP
+#define CSP_WMS_OVERLAYS_PLUGIN_HPP
+
+#include "utils.hpp"
+
+#include "../../../src/cs-core/PluginBase.hpp"
+#include "../../../src/cs-utils/DefaultProperty.hpp"
+
+#include <chrono>
+#include <map>
+#include <string>
+
+namespace csp::wmsoverlays {
+
+class TextureOverlayRenderer;
+class WebMapService;
+class WebMapLayer;
+
+/// This plugin provides the rendering of planets as spheres with a texture and an additional WMS
+/// based texture. Despite its name it can also render moons :P. It can be configured via the
+/// applications config file. See README.md for details.
+class Plugin : public cs::core::PluginBase {
+ public:
+  /// The startup settings of the plugin.
+  struct Settings {
+    /// Specifies whether to interpolate textures between timesteps (does not work when pre-fetch is
+    /// inactive).
+    cs::utils::DefaultProperty<bool> mEnableInterpolation{true};
+
+    /// Specifies whether to autmatically update the overlay bounds when the observer stopped moving
+    /// for a certain amount of time.
+    cs::utils::DefaultProperty<bool> mEnableAutomaticBoundsUpdate{false};
+
+    /// Path to the map cache folder, can be absolute or relative to the cosmoscout executable.
+    cs::utils::DefaultProperty<std::string> mMapCache{"texture-cache"};
+
+    /// Path to the wms capability cache folder, can be absolute or relative to the cosmoscout
+    /// executable.
+    cs::utils::DefaultProperty<std::string> mCapabilityCache{"wms-capability-cache"};
+
+    /// The amount of textures that gets pre-fetched in every time direction.
+    cs::utils::DefaultProperty<int> mPrefetchCount{0};
+
+    /// The size of the requested map textures along the longer axis. Some wms layers may only be
+    /// available in certain sizes, those won't be influenced by this setting.
+    cs::utils::DefaultProperty<int> mMaxTextureSize{1024};
+
+    /// If automatic bounds update is enabled, the bounds will be updated when the observer stopped
+    /// moving for this amount of milliseconds.
+    cs::utils::DefaultProperty<int> mUpdateBoundsDelay{1000};
+
+    /// The startup settings for a planet.
+    struct Body {
+      cs::utils::DefaultProperty<std::string> mActiveServer{
+          "None"}; ///< The name of the currently active WMS server.
+      cs::utils::DefaultProperty<std::string> mActiveLayer{
+          "None"}; ///< The name of the currently active WMS layer.
+      cs::utils::DefaultProperty<std::string> mActiveStyle{
+          ""}; ///< The name of the style for the currently active WMS layer.
+      cs::utils::DefaultProperty<Bounds> mActiveBounds{
+          {-180., 180., -90., 90.}}; ///< The bounds for the currently active WMS layer.
+      std::vector<std::string> mWms; ///<	URLs of WMS servers.
+    };
+
+    std::map<std::string, Body> mBodies; ///< A list of bodies with their anchor names.
```

Comment:
I would move the comments above the corresponding lines. This is more clang-format-friendly.